compiler/utils/astrepr.nim

```diff
@@ -122,124 +200,175 @@ proc format[I](s: set[I], sub: int): string =
 
   result.add "}"
 
-const setStyle = termFg(2, 1, 3)
+func format(i: SomeInteger): ColText = $i + style.number
+
+proc ownerChain(sym: PSym): string =
+  var own: seq[PSym] = @[sym]
+  var sym = sym
+  while not sym.owner.isNil():
+    sym = sym.owner
+    own.add sym
 
-func format(i: SomeInteger): ColText = $i + fgCyan
+  for idx in countdown(own.high, 0):
+    if idx < own.high:
+      result.add "."
+
+    result.add own[idx].name.s
+    result.add "("
+    result.add substr($own[idx].kind, 2)
+    result.add ")"
+
+func defaulted(r: TReprConf): bool = trfShowDefaultedFields in r
 
 proc symFields(
     conf: ConfigRef,
     sym: PSym,
-    flags: set[TReprFlag] = defaultTreeReprFlags,
+    rconf: TReprConf = implicitTReprConf,
     indent: int = 0
   ): ColText =
 
   coloredResult(1)
   var res = addr result
-  genFields(res[], indent, flags)
+  genFields(res[], indent, rconf)
 
-  if trfShowSymFlags in flags and sym.flags.len > 0:
-    field("flags", format(sym.flags, 2) + setStyle)
+  if trfShowSymFlags in rconf and
+     (sym.flags.len > 0 or rconf.defaulted()):
+    field("flags", format(sym.flags, 2) + style.setIt)
 
-  if trfShowSymMagic in flags and sym.magic != mNone:
-    field("magic", substr($sym.magic, 1) + setStyle)
+  if trfShowSymMagic in rconf and
+     (sym.magic != mNone or rconf.defaulted()):
+    field("magic", substr($sym.magic, 1) + style.setIt)
 
-  if trfShowSymOptions in flags and 0 < sym.options.len:
-    field("opts", format(sym.options, 3) + setStyle)
+  if trfShowSymOptions in rconf and
+     (0 < sym.options.len or rconf.defaulted()):
+    field("opts", format(sym.options, 3) + style.setIt)
 
-  if trfShowSymOffset in flags and 0 <= sym.offset:
-    field("offset", sym.offset.format())
+  if trfShowSymOffset in rconf and
+     (0 <= sym.offset or rconf.defaulted()):
+    field("offset", $sym.offset + style.number)
 
-  if trfShowSymPosition in flags and sym.position != 0:
-    field("pos", sym.offset.format())
+  if trfShowSymPosition in rconf and
+     (sym.position != 0 or rconf.defaulted()):
+    field("pos", $sym.offset + style.number)
 
-  if trfShowSymTypes in flags and not sym.typ.isNil():
+  if trfShowSymTypes in rconf and not sym.typ.isNil():
     field("typ", conf.textRepr(sym.typ))
 
-  if not isNil(conf) and trfShowSymLineInfo in flags and sym.info != unknownLineInfo:
-    field("info", conf.toMsgFilename(sym.info.fileIndex) + fgBlue)
-    add "(", $sym.info.line + fgCyan
-    add ".", $sym.info.col + fgCyan
+  if not isNil(conf) and
+     trfShowSymLineInfo in rconf and
+     (sym.info != unknownLineInfo or rconf.defaulted()):
+    field("info", conf.toMsgFilename(sym.info.fileIndex) + style.ident)
+    add "(", $sym.info.line + style.number
```

Comment:
Seeing similar patterns, maybe not enough occurrences, but would it make sense to create a string format template or two to absorb this?

No worries if not.